### ISSUE TITLE
bring back the "saw" IO option

### ISSUE THREAD

**Issue by lwhite1** [jtablesaw/tablesaw#619]
For those who don't remember, "saw" was a homegrown output option that was removed when we were making a lot of changes to the column implementation.

If you chose saw as your output form when you wrote a table, it produced a folder named "saw" with a metadata file (json) and n binary files, one for each column in the table. It's more like the disk representation of a postgres db than a csv file.

The advantage of the format is that it was an order-of-magnitude faster, and used less memory, than csv when reading and writing. I wanted to replace it with something similar, but more standardized and robust: like Parquet, for example. I have not been able to find anything that is pure java, however, and prefer not to have to install a separate program. 

The real use-case is this: In Python and R, reading files is slow, but in a REPL environment, you only have to read the file once per session. In java, reading files is relatively fast, but in pre-repl java, every time you want to adjust your analysis and re-run, you have to read the file again. Writing your code as regular programs is much easier in java than using the repl anyway. 

_My_ real use-case is that I regularly work with a monthly report aggregating research findings that is 1.4 GB and growing fairly rapidly. It is very string-heavy. Saw can write the String columns out in dictionary format so it doesn't need to build the three internal dictionaries when reading.  

So, saw output is likely still much faster than our much improved csv reading, even if you give the CSV reader the column types. (and it doesn't require the extra step of you providing the column types). The total output is smaller as well, and it doesn't require you to take the extra step of compressing. 

My proposal is to add saw as a subproject like json, and add it to the output format types on table.read() and table.write(). The format will be versioned so it can be changed, with the version written to the metadata file. From what i can see, there is not a lot of work here.

**Comment by benmccann** [comment]
> I wanted to replace it with something similar, but more standardized and robust: like Parquet, for example. I have not been able to find anything that is pure java, however, and prefer not to have to install a separate program.

I do think it'd be cool to support a standardized format if we can find one that fits the bill.

Isn't Parquet pure Java? It'd probably be the best fit as it's most popular

A couple other options:
* [ORC](https://orc.apache.org/docs/core-java.html)
* HDF5 - this one requires JNI, so maybe not the best option
  * [JHDF5](https://wiki-bsse.ethz.ch/display/JHDF5/JHDF5+%28HDF5+for+Java%29) wrapper to make API easier to work with
  * [HDF Java examples](https://support.hdfgroup.org/HDF5/examples/hdf-java.html)

**Comment by lwhite1** [comment]
@benmccann You're right it is java. I was confusing it with Apache Arrow. I think this is what turned me off, originally: 

> We created Parquet to make the advantages of compressed, efficient columnar data representation available to any project in the Hadoop ecosystem.

and (coincidentally)

> Orc is "the smallest, fastest columnar storage for Hadoop workloads".

Obviously, I want nothing to do with the Hadoop ecosystem. 

It seems that you can now use Parquet without HDFS and all the hadoop baggage. That wasn't possible before. But this explanation of how to read a parquet file in pure java didn't make me feel too confident: 

https://stackoverflow.com/questions/47355038/how-to-generate-parquet-file-using-pure-java-including-date-decimal-types-an

In part because it says you should include Hadoop bundle as a dependency even if you're not using it (it has other things you need), and especially the comment at the end. 

> "I was made aware that the decimal conversion code above does not work for negative values. Please keep that in mind! "

Clearly, this is a much better approach in the long term (assuming it's actually very fast). But it's not going to get done this afternoon like the saw format would have. 

I will probably write the saw version and keep it in a private repo, and reopen the old Parquet format issue

**Comment by lwhite1** [comment]
This may be helpful. [Reading Parquet Files from a Java Application](https://www.jofre.de/?p=1459)
